nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/FetchElasticsearch5.java

```diff
@@ -0,0 +1,214 @@
+        + "transport (SSL/TLS), and the X-Pack plugin is available, secure connections can be made. This processor "
+        + "supports Elasticsearch 5.x clusters.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "filename", description = "The filename attributes is set to the document identifier"),
+        @WritesAttribute(attribute = "es.index", description = "The Elasticsearch index containing the document"),
+        @WritesAttribute(attribute = "es.type", description = "The Elasticsearch document type")
+})
+public class FetchElasticsearch5 extends AbstractElasticsearch5TransportClientProcessor {
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("All FlowFiles that are read from Elasticsearch are routed to this relationship").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("All FlowFiles that cannot be read from Elasticsearch are routed to this relationship").build();
+
+    public static final Relationship REL_RETRY = new Relationship.Builder().name("retry")
+            .description("A FlowFile is routed to this relationship if the document cannot be fetched but attempting the operation again may succeed")
+            .build();
+
+    public static final Relationship REL_NOT_FOUND = new Relationship.Builder().name("not found")
+            .description("A FlowFile is routed to this relationship if the specified document does not exist in the Elasticsearch cluster")
+            .build();
+
+    public static final PropertyDescriptor DOC_ID = new PropertyDescriptor.Builder()
+            .name("el5-fetch-doc-id")
+            .displayName("Document Identifier")
+            .description("The identifier for the document to be fetched")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor INDEX = new PropertyDescriptor.Builder()
+            .name("el5-fetch-index")
+            .displayName("Index")
+            .description("The name of the index to read from")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TYPE = new PropertyDescriptor.Builder()
+            .name("el5-fetch-type")
+            .displayName("Type")
+            .description("The type of this document (used by Elasticsearch for indexing and searching)")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> relationships = new HashSet<>();
```

Comment:
It keeps coming up but we should not be building new array every time _getRelationships()_ is being called. Consider building it in static initializer once and then returning it. The same for _ getSupportedPropertyDescriptors()_.